mypy/semanal.py

```diff
@@ -1795,6 +1795,12 @@ def visit_import_from(self, imp: ImportFrom) -> None:
             missing_submodule = False
             imported_id = as_id or id
 
+            # Modules imported in a stub file without using 'from Y import X as X' will
```

Comment:
Is it possible to create a test with similar problem as in https://github.com/python/typeshed/pull/6171 ? 🤔

Reply:
Agreed. This PR looks good, but a test case would be a nice bonus.